File: 프로그래머스/level0/잘라서 배열로 저장하기.py
def solution(my_str, n):
    answer = []
    l = len(my_str)
    
    if l % n != 0:
        for i in range(l // n + 1):
            answer.append(my_str[i*n:(i+1)*n])
    else:
        for i in range(l // n):
            answer.append(my_str[i*n:(i+1)*n])
    
    return answer

File: 프로그래머스/level0/잘라서_배열로_저장하기.java
package 프로그래머스.level0;

public class 잘라서_배열로_저장하기 {
    public String[] solution(String my_str, int n) {
        int resultCnt = (my_str.length() + n - 1) / n;
        String[] answer = new String[resultCnt];

        for (int i = 0; i < resultCnt; i++) {
            int start = i * n;
            int end;
            if (start + n >= my_str.length()) {
                end = my_str.length();
            } else {
                end = start + n;
            }
            answer[i] = my_str.substring(start, end);
        }
        return answer;
    }
}
